### PR TITLE
Upgrade chain to v1.0.0

### PR DESCRIPTION
**Pull request by simonsso**
- [x]  Verify merges
- [x]  Upgrade spec_version in runtimes/eden/src/version.rs
- [x]  cargo build --release
- [x]  cargo test
- [x]  Benchmarks compiles
- [x]  Weights update or ignored
- [x]  try-runtime
- [x]  Run local test-net
- [x] Remote TODO
- [x] Restore removed uniqes-pallet

### CODE REVIEW

File: runtimes/eden/src/lib.rs [hunk 1]
```diff
@@ -432,7 +480,7 @@ sp_api::impl_runtime_apis! {
 			// have a backtrace here. If any of the pre/post migration checks fail, we shall stop
 			// right here and right now.
 			log::debug!("on_runtime_upgrade");
-			let weight = Executive::try_runtime_upgrade(checks).unwrap();
+			let weight = Executive::try_runtime_upgrade(checks).unwrap_or_default();
```
Comment (simonsso) [inline runtimes/eden/src/lib.rs:483]: This must not be here.

Comment (simonsso) [inline runtimes/eden/src/lib.rs:483]:
```suggestion
			let weight = Executive::try_runtime_upgrade(checks).unwrap();
```

File: node/src/command.rs [hunk 2]
```diff
@@ -274,9 +266,9 @@ pub fn run() -> Result<()> {
 				let id = ParaId::from(para_id);
 
 				let parachain_account =
-					AccountIdConversion::<polkadot_primitives::v4::AccountId>::into_account_truncating(&id);
+					AccountIdConversion::<polkadot_primitives::v5::AccountId>::into_account_truncating(&id);
 
-				let state_version = Cli::native_runtime_version(&config.chain_spec).state_version();
+				let state_version = StateVersion::V1; // TODO Fixme
```
Comment (aliXsed) [inline node/src/command.rs:271]: Just a reminder comment so we don't forget this

File: node/src/service.rs [hunk 3]
```diff
@@ -66,6 +66,12 @@ type ParachainBlockImport<RuntimeApi> =
 	TParachainBlockImport<Block, Arc<ParachainClient<RuntimeApi>>, ParachainBackend>;
 
 pub struct TemplateRuntimeExecutor;
```
Comment (aliXsed) [inline node/src/service.rs:68]: Is this used anymore?

File: node/src/service.rs [hunk 4]
```diff
@@ -66,6 +66,12 @@ type ParachainBlockImport<RuntimeApi> =
...
+// impl sc_executor::sp_wasm_interface::HostFunctions for TemplateRuntimeExecutor {
+
+// 	fn host_functions() -> Vec<&'static (dyn Function + 'static)> { todo!() }
+// 	fn register_static<T>(_: &mut T) -> Result<(), <T as HostFunctionRegistry>::Error> where T: HostFunctionRegistry { todo!() }
+// }
+
```
Comment (aliXsed) [inline node/src/service.rs:74]: Another cleanup reminder

File: node/src/service.rs [hunk 5]
```diff
@@ -255,6 +261,8 @@ where
 	let prometheus_registry = parachain_config.prometheus_registry().cloned();
 	let transaction_pool = params.transaction_pool.clone();
 	let import_queue_service = params.import_queue.service();
+	// TODO there should be more here:
```
Comment (aliXsed) [inline node/src/service.rs:264]: more of?

File: pallets/allocations/src/tests.rs [hunk 6]
```diff
@@ -97,11 +87,14 @@ impl pallet_balances::Config for Test {
 	type MaxReserves = ();
 	type ReserveIdentifier = [u8; 8];
 	type FreezeIdentifier = [u8; 8];
-	type HoldIdentifier = [u8; 8];
+	// type HoldIdentifier = [u8; 8];
```
Comment (aliXsed) [inline pallets/allocations/src/tests.rs:90]: cleanup reminder

File: pallets/grants/src/mock.rs [hunk 7]
```diff
@@ -93,9 +88,10 @@ impl pallet_balances::Config for Test {
 	type ReserveIdentifier = [u8; 8];
 	type WeightInfo = ();
 	type FreezeIdentifier = [u8; 8];
-	type HoldIdentifier = [u8; 8];
+	// type HoldIdentifier = [u8; 8];
```
Comment (aliXsed) [inline pallets/grants/src/mock.rs:91]: Cleanup reminder

File: pallets/grants/src/mock.rs [hunk 8]
```diff
@@ -54,25 +49,25 @@ impl frame_system::Config for Test {
...
 	type AccountData = pallet_balances::AccountData<u64>;
 	type OnNewAccount = ();
 	type OnKilledAccount = ();
+	type Block = Block;
 	type DbWeight = ();
 	type BaseCallFilter = frame_support::traits::Everything;
 	type OnSetCode = ();
 	type SystemWeightInfo = ();
 	type MaxConsumers = frame_support::traits::ConstU32<16>;
+	#[doc = " This stores the number of previous transactions associated with a sender account."]
```
Comment (aliXsed) [inline pallets/grants/src/mock.rs:69]: do we need doc for test?

File: pallets/allocations/src/tests.rs [hunk 9]
```diff
@@ -97,11 +87,14 @@ impl pallet_balances::Config for Test {
...
 	type MaxHolds = ();
 	type MaxFreezes = ();
 
 	type WeightInfo = ();
+
+	#[doc = " The overarching hold reason."]
```
Comment (aliXsed) [inline pallets/allocations/src/tests.rs:96]: No need for doc in the test

File: pallets/reserve/src/tests.rs [hunk 10]
```diff
@@ -24,22 +24,18 @@ use frame_support::{assert_noop, assert_ok, ord_parameter_types, parameter_types
...
 use sp_runtime::{
-	testing::Header,
 	traits::{BlakeTwo256, IdentityLookup},
+	BuildStorage,
 	DispatchError::BadOrigin,
 };
 use sp_std::prelude::Box;
 
-type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<Test>;
+// type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<Test>;
```
Comment (aliXsed) [inline pallets/reserve/src/tests.rs:33]: cleanup reminder

File: pallets/reserve/src/tests.rs [hunk 11]
```diff
@@ -88,9 +83,10 @@ impl pallet_balances::Config for Test {
 	type ReserveIdentifier = [u8; 8];
 	type WeightInfo = ();
 	type FreezeIdentifier = [u8; 8];
-	type HoldIdentifier = [u8; 8];
+	// type HoldIdentifier = [u8; 8];
```
Comment (aliXsed) [inline pallets/reserve/src/tests.rs:86]: cleanup reminder

File: pallets/sponsorship/src/mock.rs [hunk 12]
```diff
@@ -87,8 +81,9 @@ impl pallet_balances::Config for Test {
 	type WeightInfo = ();
 	type FreezeIdentifier = ();
 	type MaxFreezes = ();
-	type HoldIdentifier = ();
 	type MaxHolds = ();
+	#[doc = " The overarching hold reason."]
```
Comment (aliXsed) [inline pallets/sponsorship/src/mock.rs:85]: remove this line

File: pallets/uniques/src/benchmarking.rs [hunk 13]
```diff
@@ -35,7 +35,7 @@ fn get_config<T: Config<I>, I: 'static>() -> (T::CollectionId, T::AccountId, Acc
 	let collection_owner: T::AccountId = account("collection_owner", 0, SEED);
 	let collection_owner_lookup = T::Lookup::unlookup(collection_owner.clone());
 	let collection_id = <T as pallet_uniques::Config<I>>::Helper::collection(0);
-	(collection_id, collection_owner, collection_owner_lookup)
+	(collection_id.clone(), collection_owner, collection_owner_lookup)
```
Comment (aliXsed) [inline pallets/uniques/src/benchmarking.rs:38]: Why do we need to clone collection_id in this file?

File: pallets/uniques/src/lib.rs [hunk 14]
```diff
@@ -130,7 +130,7 @@ pub mod pallet {
 			collection: T::CollectionId,
 			admin: AccountIdLookupOf<T>,
 		) -> DispatchResult {
-			pallet_uniques::Pallet::<T, I>::create(origin, collection, admin)
+			pallet_uniques::Pallet::<T, I>::create(origin, collection.clone(), admin)
```
Comment (aliXsed) [inline pallets/uniques/src/lib.rs:133]: same here, why we need the clone?

File: runtimes/eden/src/pallets_consensus.rs [hunk 15]
```diff
@@ -61,14 +61,27 @@ impl pallet_aura::Config for Runtime {
...
+	#[doc = " If this is `true`, the pallet will allow slots to stay the same across sequential"]
+	#[doc = " blocks. If this is `false`, the pallet will require that subsequent blocks always have"]
+	#[doc = " higher slots than previous ones."]
+	#[doc = ""]
+	#[doc = " Regardless of the setting of this storage value, the pallet will always enforce the"]
+	#[doc = " invariant that slots don\\'t move backwards as the chain progresses."]
+	#[doc = ""]
+	#[doc = " The typical value for this should be \\'false\\' unless this pallet is being augmented by"]
+	#[doc = " another pallet which enforces some limitation on the number of blocks authors can create"]
+	#[doc = " using the same slot."]
```
Comment (aliXsed) [inline runtimes/eden/src/pallets_consensus.rs:75]: We shouldn't need docs in implementation

File: pallets/grants/src/mock.rs (hunk 8 above)
Comment (simonsso) [reply pallets/grants/src/mock.rs:69]: Maybe not, the very helpful IDE added those while I was merging it.

File: pallets/uniques/src/benchmarking.rs (hunk 13 above)
Comment (simonsso) [reply pallets/uniques/src/benchmarking.rs:38]: We need clone in some places, this was done with a find/replace to catch all.

**Comment by simonsso** [comment]
Force pushed to remove 1.1 changes (which are moved to PR #800)

File: runtimes/eden/src/weights/frame_system.rs [hunk 16]
```diff
@@ -1,115 +0,0 @@
...
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- */
-
-//! Autogenerated weights for frame_system
-//!
-//! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION 4.0.0-dev
-//! DATE: 2023-09-13, STEPS: `50`, REPEAT: 20, LOW RANGE: `[]`, HIGH RANGE: `[]`
```
Comment (simonsso) [inline runtimes/eden/src/weights/frame_system.rs:22]: This was not updated by the scripts and nothing broke when it was removed

Comment (simonsso) [reply runtimes/eden/src/weights/frame_system.rs:22]: > This was not updated by the scripts and nothing broke when it was removed


```
2023-11-27 17:50:57 assembling new collators for new session 0 at #0    
2023-11-27 17:50:57 assembling new collators for new session 1 at #0    
2023-11-27 17:50:58 Starting benchmark: frame_system::remark    
2023-11-27 17:51:01 Starting benchmark: frame_system::remark_with_event    
2023-11-27 17:51:03 Running  benchmark: frame_system.remark_with_event(1 args) 34/50 1/1    
2023-11-27 17:51:06 Starting benchmark: frame_system::set_heap_pages    
2023-11-27 17:51:06 Starting benchmark: frame_system::set_code    
Error: Input("Benchmark frame_system::set_code failed: ValidationDataNotAvailable")
```
This is maybe not the correct approach here

File: runtimes/eden/src/migrations.rs [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+		// for Polkadot which got inserted in 3614349 16 days ago. I believe we don’t need this migration.
+		// If in the future there was any issue in any XCM interactions with Moonbeam we can force set the
+		// storage entry for that single value to use proof_size = 65536 (the new default).
+		StorageVersion::new(1).put::<pallet_xcm::Pallet<T>>();
+
+		// Size of onchain storage is 0 safe to upgrade storage version
+		StorageVersion::new(1).put::<pallet_preimage::Pallet<T>>();
+
+		// give it some weight not zero
+		Weight::from_parts(10000, 0x10000)
```
Comment (aliXsed) [inline runtimes/eden/src/migrations.rs:52]: Just a reminder to return a better estimate of weight here

Comment (simonsso) [reply runtimes/eden/src/migrations.rs:69]: That would be the weight of 5 single write operations.

File: runtimes/eden/src/xcm_config.rs [hunk 18]
```diff
@@ -159,6 +159,8 @@ impl xcm_executor::Config for XcmConfig {
 	type UniversalAliases = Nothing;
 	type CallDispatcher = RuntimeCall;
 	type SafeCallFilter = Everything;
+
+	type Aliasers = Nothing; // TODO check if it should be type Aliasers = AliasForeignAccountId32<OnlyParachains>;
```
Comment (simonsso) [inline runtimes/eden/src/xcm_config.rs:163]: @aliXsed One todo left to investigate here.

File: runtimes/eden/src/migrations.rs [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+		+ pallet_balances::Config
+		+ pallet_membership::Config<pallet_membership::pallet::Instance3>
+		+ pallet_collective::Config<pallet_collective::pallet::Instance1>
+		+ pallet_collator_selection::Config
+		+ pallet_xcm::Config
+		+ pallet_preimage::Config
+		+ pallet_multisig::Config
+		+ pallet_contracts::Config,
+{
+	fn on_runtime_upgrade() -> Weight {
```
Comment (simonsso) [inline runtimes/eden/src/migrations.rs:22]: @aliXsed please check if the reasoning below holds.

File: runtimes/eden/src/constants.rs [hunk 20]
```diff
@@ -73,7 +74,7 @@ pub const NORMAL_DISPATCH_RATIO: Perbill = Perbill::from_percent(75);
 /// We allow for .5 seconds of compute with a 16 second average block time.
 pub const MAXIMUM_BLOCK_WEIGHT: Weight = Weight::from_parts(
 	WEIGHT_REF_TIME_PER_SECOND.saturating_div(2),
-	polkadot_primitives::v4::MAX_POV_SIZE as u64,
+	polkadot_primitives::v5::MAX_POV_SIZE as u64,
```
Comment (aliXsed) [inline runtimes/eden/src/constants.rs:77]: import MAX_POV_SIZE from a place which points ti the last version automatically

File: runtimes/eden/src/lib.rs [hunk 21]
```diff
@@ -136,6 +137,8 @@ pub type Block = generic::Block<Header, UncheckedExtrinsic>;
 pub type SignedBlock = generic::SignedBlock<Block>;
 /// BlockId type as expected by this runtime.
 pub type BlockId = generic::BlockId<Block>;
+/// Index of a transaction in the chain.
+pub type Nonce = u32;
```
Comment (aliXsed) [inline runtimes/eden/src/lib.rs:141]: Suggest moving to primitives

File: runtimes/eden/src/migrations.rs (hunk 19 above)
Comment (aliXsed) [reply runtimes/eden/src/migrations.rs:22]: @simonsso for those you didn't find any keys and you have only changed the storage version, you still need to run the migration logic if they have any. The reason being is we cannot guarantee there wouldn't be any keys inserted for those pallets before we run the actual upgrade.

Comment (aliXsed) [reply runtimes/eden/src/migrations.rs:22]: Unless the version deployed on mainnet is already the very latest for that pallet

File: runtimes/eden/src/migrations.rs [hunk 22]
```diff
@@ -0,0 +1,103 @@
...
+		// because two new values of the new type were inserted in the VersionNotifyTargets map which is
+		// the subject of that migration. One of the new values are for Moonbeam which got inserted in
+		// the block 3351853 which is the first block after the parachain restart and the second one is
+		// for Polkadot which got inserted in 3614349 16 days ago. I believe we don’t need this migration.
+		// If in the future there was any issue in any XCM interactions with Moonbeam we can force set the
+		// storage entry for that single value to use proof_size = 65536 (the new default).
+		StorageVersion::new(1).put::<pallet_xcm::Pallet<T>>();
+
+		// Size of onchain storage is 0 safe to upgrade storage version
+		StorageVersion::new(1).put::<pallet_preimage::Pallet<T>>();
```
Comment (aliXsed) [inline runtimes/eden/src/migrations.rs:87]: Just double check the deployed code is the latest version

File: runtimes/eden/src/xcm_config.rs [hunk 23]
```diff
@@ -190,6 +191,8 @@ impl pallet_xcm::Config for Runtime {
 	type WeightInfo = crate::weights::pallet_xcm::WeightInfo<Runtime>;
 	#[cfg(feature = "runtime-benchmarks")]
 	type ReachableDest = ReachableDest;
+	type MaxRemoteLockConsumers = ConstU32<0>;
+	type RemoteLockConsumerIdentifier = ();
```
Comment (aliXsed) [inline runtimes/eden/src/xcm_config.rs:195]: Can we double check these

File: runtimes/eden/src/migrations.rs (hunk 19 above)
Comment (simonsso) [reply runtimes/eden/src/migrations.rs:22]: Checked all the pallets in substrate. XCM had been checked manually before.

File: runtimes/eden/src/constants.rs (hunk 20 above)
Comment (simonsso) [reply runtimes/eden/src/constants.rs:77]: done

File: runtimes/eden/src/xcm_config.rs (hunk 23 above)
Comment (simonsso) [reply runtimes/eden/src/xcm_config.rs:195]: https://github.com/paritytech/polkadot/pull/6947

https://github.com/paritytech/polkadot/pull/6947#issuecomment-1527287885

This is used in all applications and we don't use it either

**Pull request merged**